SteamKit2/SteamKit2/Types/Manifest.cs

```diff
@@ -110,33 +110,17 @@ internal void Deserialize( BinaryReader ds )
         [NotNull]
         public List<FileMapping>? Mapping { get; private set; }
 
-        public Steam3Manifest(byte[] data)
-        {
-            ArgumentNullException.ThrowIfNull( data );
-
-            Deserialize(data);
-        }
-
-        internal Steam3Manifest(BinaryReader data)
-        {
-            Deserialize(data);
-        }
-
-        void Deserialize(byte[] data)
-        {
-            using var ms = new MemoryStream( data );
-            using var br = new BinaryReader( ms );
-            Deserialize( br );
-        }
-
-        void Deserialize( BinaryReader ds )
+        internal void Deserialize( BinaryReader ds )
         {
+            /*
+            // The magic is verified by DepotManifest.InternalDeserialize, not checked here to avoid seeking

```

Comment:
This class is not public, and deserialize is only invoked from one code path, so it's fine.